packages/react/src/SubdomainNavBar/SubdomainNavBar.tsx

```diff
@@ -266,7 +267,75 @@ const _SearchInternal = (
 ) => {
   const dialogRef = useRef<HTMLDivElement | null>(null)
 
-  useOnClickOutside(dialogRef, handlerFn)
+  const [activeDescendant, setActiveDescendant] = useState<number>(-1)
+  const [listboxActive, setListboxActive] = useState<boolean>()
+  const [liveRegion, setLiveRegion] = useState<boolean>(false)
+
+  const handleClose = event => {
+    if (handlerFn) handlerFn(event)
+    setActiveDescendant(-1)
+  }
+
+  useOnClickOutside(dialogRef, handleClose)
+  useKeyboardEscape(() => {
+    setListboxActive(false)
+    setActiveDescendant(-1)
+  })
+
+  const handleAriaFocus = event => {
+    const supportedKeys = ['ArrowDown', 'ArrowUp', 'Escape', 'Enter']
+    const currentCount = activeDescendant
+    const searchResultsLength = searchResults ? searchResults.length : 0
+    const dialog = dialogRef.current
+    let count
+
+    // Prevent any other keys outside of supported from being prevented.
+    // Only prevent "Enter" if activeDescendant is greater than -1.
+    if (!supportedKeys.includes(event.key) || (event.key === 'Enter' && activeDescendant === -1) || !dialog) {
+      return false
+    }
+
+    event.preventDefault()
```

Comment:
Could this be hoisted to the top of the handler, or are you partially relying on some default behavior?

Reply:
I put it below so that it wouldn't restrict regular key presses outside of arrow keys, and the enter key in some scenarios. Mainly so users can still type without it being prevented.